openpype/plugins/publish/integrate_inputlinks_ayon.py

```diff
@@ -144,16 +175,30 @@ def create_links_on_server(self, context, new_links):
 
         # Create link themselves
         for link_type, items in new_links.items():
+            mapping = collections.defaultdict(set)
+            # Make sure there are no duplicates of src > dst ids
             for item in items:
-                input_id, output_id = item
-                create_link(
-                    project_name,
-                    link_type,
-                    input_id,
-                    "version",
-                    output_id,
-                    "version"
-                )
+                _input_id, _output_id = item
+                mapping[_input_id].add(_output_id)
+
+            existing_links_by_in_id = self._get_existing_links(
+                project_name, link_type, set(mapping.keys())
+            )
+
+            for input_id, output_ids in mapping.items():
+                existing_links = existing_links_by_in_id[input_id]
+                for output_id in output_ids:
+                    # Skip creation of link if already exists
+                    if output_id in existing_links:
+                        continue
+                    create_link(
+                        project_name,
+                        link_type,
+                        input_id,
+                        "version",
+                        output_id,
+                        "version"
+                    )
```

Comment:
Question!
I have a feeling that creating each link individually on the backend as a separate creation request would be massively slow, no? You can easily have 10s or 100s of links per instance. With say 10 instances that's doing quite a few calls to the backend - or is the ayon api caching these and is a `session.commit()` or whatever needed to finish the submission?

Reply:
> Also, another question - the link is like this right output_id -> input_id? Somehow my OCD trips over the fact that you need to specify the destination first and the source last as arguments to create_link

I don't understand this comment (nothing of it), but I guess that's because I don't understand the links flow much...